### ISSUE TITLE
Unable to get Futures working concurrently

### ISSUE THREAD

**Issue by vineethvarghese** [atnos-org/eff#109]
We are unable to get `TimedFuture` to do any work concurrently after we upgraded from 3.0.4. I have created the reproducible test cases at :- 
- https://github.com/vineethvarghese/eff-parallel-4-2-0 (Not working)
- https://github.com/vineethvarghese/eff-parallel-3-0-4 (Working)

Despite having enough threads, the futures seems to be executing sequentially after we upgrade from 3.0.4 to 4.0.2. We have attached YourKit thread images in those repos.

Is this is bug or are we using the library wrong in some way?

Another point to note is that the following code where we use TimedFutures directly (rather than being translated into) works fine on 4.2.0 :- 
```scala
import scala.concurrent.Await
import scala.concurrent.duration._
import scala.concurrent.ExecutionContext.Implicits.global

import java.util.concurrent.Executors

import cats.implicits._

import org.atnos.eff._
import org.atnos.eff.Interpret.translate
import org.atnos.eff.future._
import org.atnos.eff.all._
import org.atnos.eff.syntax.all._
import org.atnos.eff.syntax.future._


object Test {
  def q(i: Int, sleep: Long) = for {
    _ <- p(i, 0, sleep)
    _ <- p(i, 1, sleep)
    _ <- p(i, 2, sleep)
  } yield ()

  def p(i: Int, j: Int, sleep: Long) = futureDelay {
    println(s"Starting $i-$j")
    Thread.sleep(sleep)
    println(s"Finished $i-$j")
  }

  type T = Fx.fx1[TimedFuture]
  val x = Eff.traverseA[T, List, Int, Unit](List(1, 2, 3, 4))(i => q(i, 10000))

  def main(args: Array[String]): Unit = {
    implicit val scheduler = ExecutorServices.schedulerFromScheduledExecutorService(Executors.newSingleThreadScheduledExecutor())
    Await.result(
      x.runAsync,
      5 minutes
    )
  }
}
```

**Comment by etorreborre** [comment]
I think I have pinpointed where the problem is and why it was introduced. It was in making `traverseA` stacksafe. If I revert one line I get back the concurrency. But I lose the stack-safety :-(. It is not yet clear to me how to fix this to retain both. I'll keep you updated.

**Comment by vineethvarghese** [comment]
Thank you @etorreborre. So, TimeFuture in 3.0.4(with traverseA) is not stacksafe but it works concurrently. :) We reverted to 3.0.4 because concurrency was the only reason we went to TimeFutures.

**Comment by etorreborre** [comment]
I have just released `4.4.0` with a fix. It should fix your case but I don't think that's the end of the story though.

**Comment by vineethvarghese** [comment]
@etorreborre That seems to have worked. Thank you. We might have to test out the 100 effects limit to understand how it will affect us (if not now, then later).

![eff-parallel-4-4-0-assembly-0_1-snapshot_-_yourkit_java_profiler_2017_02-b59_-_64-bit](https://cloud.githubusercontent.com/assets/672908/26813767/ed7d715e-4ac3-11e7-84e9-09a32071b248.png)
